### ISSUE TITLE
`supervisorctl tail -100 myprog stderr` command does not work

### ISSUE THREAD

**Issue by timotheecour** [Supervisor/supervisor#1293]
(on OSX if that matters)

this works:
```
supervisorctl tail -f myprog stderr
supervisorctl tail -f myprog stdout
supervisorctl tail -100 myprog # for stdout
```

this does not work:
```
supervisorctl tail -100 myprog stderr
error: <class 'xml.parsers.expat.ExpatError'>, not well-formed (invalid token): line 5, column 25: file: /Users/timothee/homebrew/opt/python@2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xmlrpclib.py line: 558
```

even though it should according to docs:

```
supervisorctl help tail
tail [-f] <name> [stdout|stderr] (default stdout)
Ex:
tail -f <name>          Continuous tail of named process stdout
                        Ctrl-C to exit.
tail -100 <name>        last 100 *bytes* of process stdout
tail <name> stderr      last 1600 *bytes* of process stderr
```

so there's no way to retrieve past stderr (only live stderr via tail -f) which makes it hard to diagnose issues after the fact

/cc @mnaberez  because you wrote https://github.com/Supervisor/supervisor/issues/1050#issuecomment-362091286 even this cmd does not work for me (on OSX at least)

**Comment by mnaberez** [comment]
Here's what I tried to reproduce this.

Python script `spew.py` that writes to `stderr`:
```
import sys
import time

while True:
  sys.stderr.write("this is stderr ")
  sys.stderr.flush()
  time.sleep(0.1)
```

Minimal `supervisord.conf` to run it:
```
[supervisord]

[inet_http_server]
port = 127.0.0.1:9001

[supervisorctl]
serverurl = http://127.0.0.1:9001

[rpcinterface:supervisor]
supervisor.rpcinterface_factory = supervisor.rpcinterface:make_main_rpcinterface

[program:spew]
command = python spew.py
startsecs = 0
autorestart = false
```

Start `supervisord` in the foreground with that config:
```
$ supervisord --version
4.1.0
$ supervisord -n -c supervisord.conf
2019-11-10 10:58:10,336 INFO Increased RLIMIT_NOFILE limit to 1024
2019-11-10 10:58:10,344 INFO RPC interface 'supervisor' initialized
2019-11-10 10:58:10,344 CRIT Server 'inet_http_server' running without any HTTP authentication checking
2019-11-10 10:58:10,345 INFO supervisord started with pid 25247
2019-11-10 10:58:11,352 INFO spawned: 'spew' with pid 25252
2019-11-10 10:58:11,388 INFO success: spew entered RUNNING state, process has stayed up for > than 0 seconds (startsecs)
```

In another terminal, try `supervisorctl`:
```
$ supervisorctl -c  supervisord.conf 
spew                             RUNNING   pid 25252, uptime 0:00:52

supervisor> tail -100 spew stderr
is stderr this is stderr this is stderr this is stderr this is stderr this is stderr this is stderr 
supervisor> 
```

> so there's no way to retrieve past stderr (only live stderr via tail -f) which makes it hard to diagnose issues after the fact

It works in at least some cases as shown above.  There may be something about the contents of your `stderr` logfile that are causing it trouble.  We'd need more information to troubleshoot.

**Comment by mnaberez** [comment]
Closing after 30 days of inactivity.  We'd need more information to troubleshoot.

However, the error above ...
> error: <class 'xml.parsers.expat.ExpatError'>, not well-formed (invalid token): line 5, column 25: file: /Users/timothee/homebrew/opt/python@2/Frameworks/Python.framework/Versions/2.7/lib/python2.7/xmlrpclib.py line: 558

... also appears in #85 so it's possible that fixing #85 would fix this as well.

**Comment by timotheecour** [comment]
@mnaberez could we please re-open? I did more investigation: 
* `tail -f` works reliably with both stdout and stderr, with or without ansii escape codes (such as color codes)
* `tail -100` does not work if there are ansii escape codes; this is blocked by https://github.com/Supervisor/supervisor/issues/85 but that issue didn't mention that only `tail -100` chokes on color codes, not `tail -f`
* (that part is new and different from https://github.com/Supervisor/supervisor/issues/85): when there are no ansii escape codes, `tail -100 progname stdout` incorrectly shows stderr stream:

on a program that just prints to both stderr and stdout (using a prefix stderr: and stdout:): 

```
supervisorctl tail -100 temp2_cmd_echo_forever stderr

stderr: cmd_echo2_forever: 47397
stderr: cmd_echo2_forever: 47398
stderr: cmd_echo2_forever: 47399

supervisorctl tail -100 temp2_cmd_echo_forever stdout

stderr: cmd_echo2_forever: 47633
stderr: cmd_echo2_forever: 47634
stderr: cmd_echo2_forever: 47635

supervisorctl tail -100 temp2_cmd_echo_forever
orever: 47500
stdout: cmd_echo2_forever: 47501
stdout: cmd_echo2_forever: 47502
stdout: cmd_echo2_fo
```

as a workaround, use `supervisorctl tail -100 progname` instead of `supervisorctl tail -100 progname stdout`

**Comment by mnaberez** [comment]
> tail -100 progname stdout incorrectly shows stderr stream

Fixed in #1252
